### PR TITLE
include the routing module

### LINKED ISSUE #24: add hashrouting

### PR DESCRIPTION
**Pull request by s-cork**
- [x] fix formatting

closes #24

### CODE REVIEW

**Comment by s-cork** [comment]
can we change the max line length for black - i really don't like 79... I personally prefer 119.

**Comment by meatballs** [comment]
Black uses 88 https://black.readthedocs.io/en/stable/the_black_code_style.html

On March 13, 2021 4:22:17 AM UTC, stu ***@***.***> wrote:
>can we change the max line length for black - i really don't like 79...
>I personally prefer 119.

-- 
Owen Campbell 

This email has been digitally signed. 
For further information visit http://owencampbell.me.uk/pgp

**Comment by s-cork** [comment]
I'll make the docs a separate pr I think.
Let me know if there's anything you want changing in the code or questions.

File: client_code/routing/_navigation.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
+from time import sleep
```
Comment (meatballs) [inline client_code/routing/_navigation.py:1]: Licence text here too

File: client_code/routing/_routing.py [hunk 2]
```diff
@@ -0,0 +1,751 @@
+from collections import namedtuple as _namedtuple
```
Comment (meatballs) [inline client_code/routing/_routing.py:1]: licence text here too

File: client_code/routing/_session_expired.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+from anvil.js.window import jQuery as _S
```
Comment (meatballs) [inline client_code/routing/_session_expired.py:1]: licence text

**Review by meatballs — CHANGES_REQUESTED** [review]

**Comment by meatballs** [comment]
Let me push a fix for the github action. I hit this yesterday on another repo and forgot it would apply here too.

**Review by meatballs — APPROVED** [review]

**Comment by meatballs** [comment]
Nice! That's a big step forward for this repo!

**Pull request merged**